radio/src/targets/taranis/hal.h

```diff
@@ -1822,91 +1827,57 @@
-  #define TRAINER_IN_CCMR1              TIM_CCMR1_IC2F_0 | TIM_CCMR1_IC2F_1 | TIM_CCMR1_CC2S_0;
-  #define TRAINER_OUT_COUNTER_REGISTER  TRAINER_TIMER->CCR1
-  #define TRAINER_IN_COUNTER_REGISTER   TRAINER_TIMER->CCR2
-  #define TRAINER_SETUP_REGISTER        TRAINER_TIMER->CCR3
-  #define TRAINER_OUT_INTERRUPT_FLAG    TIM_SR_CC3IF
-  #define TRAINER_OUT_INTERRUPT_ENABLE  TIM_DIER_CC3IE
-  #define TRAINER_IN_INTERRUPT_ENABLE   TIM_DIER_CC2IE
-  #define TRAINER_IN_INTERRUPT_FLAG     TIM_SR_CC2IF
-  #define TRAINER_OUT_CCER              TIM_CCER_CC1E
-  #define TRAINER_CCER_POLARYTY         TIM_CCER_CC1P
-  #define TRAINER_IN_CCER               TIM_CCER_CC2E
 #elif defined(PCBXLITE)
   #define TRAINER_TIMER                 TIM4
   #define TRAINER_RCC_AHB1Periph        0
-  #define TRAINER_RCC_APB1Periph        RCC_APB1Periph_TIM4
   #define TRAINER_TIMER_IRQn            TIM4_IRQn
   #define TRAINER_TIMER_IRQHandler      TIM4_IRQHandler
 #else
   #define TRAINER_RCC_AHB1Periph        (RCC_AHB1Periph_GPIOA | RCC_AHB1Periph_GPIOC | RCC_AHB1Periph_DMA1)
-  #define TRAINER_RCC_APB1Periph        RCC_APB1Periph_TIM3
   #define TRAINER_GPIO                  GPIOC
-  #define TRAINER_IN_GPIO_PIN           GPIO_Pin_8  // PC.08
-  #define TRAINER_IN_GPIO_PinSource     GPIO_PinSource8
-  #define TRAINER_OUT_GPIO_PIN          GPIO_Pin_9  // PC.09
-  #define TRAINER_OUT_GPIO_PinSource    GPIO_PinSource9
+  #define TRAINER_IN_GPIO_PIN           LL_GPIO_PIN_8  // PC.08
+  #define TRAINER_IN_TIMER_Channel      LL_TIM_CHANNEL_CH3
+  #define TRAINER_OUT_GPIO_PIN          LL_GPIO_PIN_9  // PC.09
+  #define TRAINER_OUT_TIMER_Channel     LL_TIM_CHANNEL_CH4
   #define TRAINER_DETECT_GPIO           GPIOA
-  #define TRAINER_DETECT_GPIO_PIN       GPIO_Pin_8  // PA.08
-#if defined(RADIO_X9DP2019) || defined(RADIO_X7ACCESS)
-  #define TRAINER_DETECT_GPIO_PIN_VALUE Bit_SET
-#else
-  #define TRAINER_DETECT_GPIO_PIN_VALUE Bit_RESET
+  #define TRAINER_DETECT_GPIO_PIN       LL_GPIO_PIN_8  // PA.08
+#if !defined(RADIO_X9DP2019) && !defined(RADIO_X7ACCESS)
+  #define TRAINER_DETECT_INVERTED
 #endif
   #define TRAINER_TIMER                 TIM3
   #define TRAINER_TIMER_IRQn            TIM3_IRQn
-  #define TRAINER_GPIO_AF               GPIO_AF_TIM3
-  #define TRAINER_DMA                   DMA1
-  #define TRAINER_DMA_CHANNEL           DMA_Channel_5
-  #define TRAINER_DMA_STREAM            DMA1_Stream2
-  #define TRAINER_DMA_IRQn              DMA1_Stream2_IRQn
-  #define TRAINER_DMA_IRQHandler        DMA1_Stream2_IRQHandler
-  #define TRAINER_DMA_FLAG_TC           DMA_IT_TCIF2
+  #define TRAINER_GPIO_AF               LL_GPIO_AF_2
+  // #define TRAINER_DMA                   DMA1
+  // #define TRAINER_DMA_CHANNEL           DMA_Channel_5
+  // #define TRAINER_DMA_STREAM            DMA1_Stream2
+  // #define TRAINER_DMA_IRQn              DMA1_Stream2_IRQn
+  // #define TRAINER_DMA_IRQHandler        DMA1_Stream2_IRQHandler
+  // #define TRAINER_DMA_FLAG_TC           DMA_IT_TCIF2
```

Comment:
are you going to need those for anything? If not, throw'em away

Reply:
These can be removed, as I can deduct them from the pins.